lib/base64.nit

```diff
@@ -77,62 +77,74 @@ redef class NativeString
 
 	# Decodes `self` from base64
 	#
+	# `strict` is the reverse-equivalent of the `-i` option of
+	# the `base64` Unix command.
+	# i.e., it will check if all characters in `self` are from the
+	# `base64` alphabet and is correctly padded if needed.
+	# `strict` defaults to `true`
+	#
 	#      assert "c3RyaW5n".decode_base64 == "string"
 	#      assert "c3Rya\nW5n".decode_base64 == "string"
+	#      assert "c3RyaW5nCg==".decode_base64 == "string\n"
+	#      assert "c3RyaW5nCg".decode_base64(strict=false) == "string\n"
+	#      assert "c3RyaW5neQo=".decode_base64 == "stringy\n"
+	#      assert "c3RyaW5neQo".decode_base64(strict=false) == "stringy\n"
 	#
-	# REQUIRE: `length % 4 == 0`
-	private fun decode_base64(length: Int, padding: nullable Byte): Bytes do
-		if padding == null then padding = '='.ascii
+	private fun decode_base64(length: Int, padding: nullable Byte, strict: nullable Bool): Bytes do
 		var inv = once inverted_base64_chars
 		if length == 0 then return new Bytes.empty
 
-		# Remove non-base64 chars
-		var bytes = new Bytes.with_capacity(length)
-		for k in [0 .. length[ do
-			var byte = self[k]
-			if inv.has_key(byte) or byte == padding then bytes.add(byte)
-		end
-		length = bytes.length
-
-		var steps = length / 4
-		var result_length = steps * 3
-
-		var epos = length - 1
-		var padding_len = 0
-		while epos >= 0 and bytes[epos] == padding do
-			epos -= 1
-			padding_len += 1
+		# Avoids constant unboxing
+		var pad = padding or else '='.ascii
+		var str = strict or else true
```

Comment:
In code that manipulate strings, what kind of sadist names `str` a local variable that contains a strictness flag?